python/deltalake/table.py

```diff
@@ -473,15 +478,48 @@ def update(
         >>> from deltalake import DeltaTable
         >>> dt = DeltaTable("tmp")
         >>> dt.update(updates = {
-        ...             "deleted": True,
+        ...             "deleted": 'True',
         ...             "id": "concat(id, '_old')"
         ...             }
         ...         )
 
+        Update some row values with python object. This is equivalent to
+        ``UPDATE table SET price = 150.10 WHERE id = '5'``
+        >>> from deltalake import DeltaTable
+        >>> dt = DeltaTable("tmp")
+        >>> dt.update(predicate="id = '5'",
+        ...           updates = {
+        ...             "price": 150.10,
+        ...             }
+        ...         )
         """
-
+        if updates is None and new_values is not None:
+            updates = {}
+            for key, value in new_values.items():
+                if isinstance(value, (int, float, bool, list)):
+                    value = str(value)
+                elif isinstance(value, str):
+                    value = f"'{value}'"
+                elif isinstance(value, datetime):
+                    value = str(
+                        int(value.timestamp() * 1000 * 1000)
+                    )  # convert to microseconds
+                updates[key] = value
+        elif updates is not None and new_values is None:
+            pass
```

Comment:
Probably worth validating here that `updates` values are all strings. Can provide a helpful message saying "The values of the updates parameter must all be SQL strings. Got {}. Did you mean to use the new_values parameter?"

Reply:
Done :)